core/__tests__/models/destination/modelCache.ts

```diff
@@ -0,0 +1,152 @@
+import { helper } from "@grouparoo/spec-helper";
+import { Destination } from "../../../src";
+import { DestinationsCache } from "../../../src/modules/caches/destinationsCache";
+
+describe("models/destinationsCache", () => {
+  helper.grouparooTestServer({ truncate: true, enableTestPlugin: true });
+
+  beforeEach(() => (DestinationsCache.TTL = 30 * 1000));
+  afterAll(() => (DestinationsCache.expires = 0));
+
+  beforeAll(async () => {
+    await helper.factories.properties();
+    const deletedDestination = await helper.factories.destination();
+    await deletedDestination.update({ state: "deleted" });
+  });
+
+  describe("#findAllWithCache", () => {
+    test("it returns all the instances", async () => {
+      const instances = await DestinationsCache.findAllWithCache();
+      expect(instances.length).toBe(1);
+    });
+
+    test("it can filter to state", async () => {
+      const destination = await helper.factories.destination();
+      expect(destination.state).toBe("ready");
+
+      const instances = await DestinationsCache.findAllWithCache(
+        undefined,
+        "ready"
+      );
+
+      expect(instances.length).toBe(1);
+      expect(instances[0].id).toBe(destination.id);
+
+      await destination.destroy();
+    });
+
+    describe("rpc", () => {
```

Comment:
nit: this `describe` block for rpc is probably outside the `#findAllWithCache` describe block.